ls1intum/Artemis
# `Quiz exercises`: Increase usability for drag and drop and short answer question creation and participation

### Checklist
#### General


#### Client

### Motivation and Context


We changed the library that realizes the drag and drop functionality for the quizzes. Therefore, still some differences are visible while creating and participating in a DnD quiz and creating a SA quiz between the old implementation and the new one. 
The old implementation provided the following visual details, the new one currently does not:

- When hovering over a drag item, the cursor changed to a pointer
- Participation in DnD questions: When dragging a drag item, the text items where still contained in a white box for the preview next to your cursor.
- When dragging picture drag items, the preview does not show the picture in full resolution, but keeps the size of the drag item.
- When a drag item is hooked by the cursor (so the user begins to drag the item), the background of the drop locations changed to blue in order to visualize that this is a possible dropping spot
- When the user hovered with a drag item over a drop location, the background of the drop location changed to green and no placeholder within the drop location was shown.

### Description

I added CSS classes that are used by the library in order to achieve the visuals explained above

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Log in to Artemis
2. Create a quiz with a DnD question
3. When creating the mappings between drag item and drop location, make sure that:

- When dragging an item, a preview of the drag item is shown next to the cursor and all possible drop locations turn blue
- For picture drag items: the preview does not render the picture in original resolution, it keeps the size of the drag item
- When hovering over a drop location while dragging an item, the drop location turns green + no placeholder is shown within the drop location

4. Also check the preview mode for the DnD question that it behaves consistently to the participation view (see list below)
5. Create a short answer question. Switch to the visual representation (button "visual" in the lower left corner of the question) and check the same things as above for this drag and drop functionality. 
**Note**: If you are testing with the default SA question, click inside the markdown editor once before switching to visual mode. Otherwise, the drag items won't get parsed. This could be solved better but is not in scope of this PR.
5. Participate in the quiz as student. Check the following things (only for the DnD question):

- When hovering over a drag item, the cursor turns into a pointer (or into move when dragging a picture drag item)
- Again, a preview is shown next to the cursor (including the white box for text drag items) while dragging and all possible drop locations turn blue
- Again, picture drag items stay in size
- Again, when hovering over a drop location while dragging an item, the drop location turns green and no placeholder is shown within the drop location. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





Nothing changed

### Screenshots


How it looks now during quiz creation:

DnD:

https://user-images.githubusercontent.com/80622272/149154017-397e1c35-c822-4c71-bec7-efa625f71966.mp4


DnD Preview mode:


https://user-images.githubusercontent.com/80622272/149153937-9fee2580-e91b-4e0a-9c40-ca8ed8cee2c1.mp4


SA:

https://user-images.githubusercontent.com/80622272/149122558-631480a8-a791-4105-a75e-83aa5fe33e4f.mp4


How it looks now during quiz participation:

https://user-images.githubusercontent.com/80622272/149154078-2be7250a-e2b7-4a55-9e67-8e9fd02d03ab.mp4
